main.tf

```diff
@@ -47,6 +48,10 @@ resource "google_cloudfunctions_function" "main" {
   event_trigger {
     event_type = "${var.event_trigger["event_type"]}"
     resource   = "${var.event_trigger["resource"]}"
+
+    failure_policy {
+      retry = "${var.event_trigger_failure_policy_retry}"
```

Comment:
I tried doing something similar to `${var.event_trigger["resource"]["failure_policy"]["retry"]}` but it failed. Is there a preferred way to reference nested maps?

Reply:
I'll give this a try.

Reply:
Coming back to this topic, I'm not sure how an array of maps would work or help here. Could you elaborate?

Reply:
`event_trigger` is a "block", which is actually equivalent to a list of maps. However, the provider [limits the list to 1 element](https://github.com/terraform-providers/terraform-provider-google/blob/v2.9.1/google/resource_cloudfunctions_function.go#L231), which means we should be able to pass it like the following without specifically fetching each key-value pair:

```hcl
resource "google_cloudfunctions_function" "main" {
  # ...
  event_trigger = ["${var.event_trigger}"]
  # ...
}
```

Reply:
I suppose I'm conflicted here. With us defining each as its own variable the user is able to look at the module's variables and know exactly how to configure it, were as if we leave `event_trigger` as a map the user then needs to lookup the format of this specific map.

Reply:
Fair. Alternatively, let's break up the `event_trigger` map variable to individual variables.